pkg/reconciler/autoscaling/kpa/scaler.go

```diff
@@ -252,6 +252,9 @@ func (ks *scaler) applyScale(ctx context.Context, pa *pav1alpha1.PodAutoscaler,
 func (ks *scaler) Scale(ctx context.Context, pa *pav1alpha1.PodAutoscaler, desiredScale int32) (int32, error) {
 	logger := logging.FromContext(ctx)
 
+	if desiredScale < 0 && pa.Status.IsInactive() {
```

Comment:
A brief comment describing how we'd get here would be great (i.e. "If the underlying application has terminally failed, scale to 0")